sycl/source/detail/scheduler/commands.cpp

```diff
@@ -243,6 +247,15 @@ static void flushCrossQueueDeps(const std::vector<EventImplPtr> &EventImpls,
   }
 }
 
+static AllocaCommandBase *
+allocaCmdGetOrThrow(std::optional<AllocaCommandBase *> allocaCmd) {
```

Comment:
Seeing as this is only called in conjunction with `getAllocaForReq`, I would suggest to replace this with `getAllocaForReqOrThrow` that would just call `getAllocaForReq` and make the check.